Client/Frontend/Settings/Main/AppSettingsTableViewController.swift

```diff
@@ -135,14 +137,15 @@ class AppSettingsTableViewController: SettingsTableViewController, AppSettingsSc
 
     func handle(route: Route.SettingsSection) {
         switch route {
-        case .creditCard:
-            handleCreditCardAuthenticatinFlow()
+        case .creditCard, .password:
+            authenticateUserFor(route: route)
```

Comment:
Seems we're missing something, since this calls `authenticateUserFor` for the `password` enum, which then possibly calls `showCreditCardSettings` when we're device owner authenticated

Reply:
It should call openDeferredRouteAfterAuthentication which will handle the specific route correctly.